src/algorithm.jl

```diff
@@ -93,7 +93,7 @@ function MathProgBase.loadproblem!(m::PavitoNonlinearModel, numvar, numconstr, l
     m.d = d
     m.vartypes = fill(:Cont, numvar)
 
-    MathProgBase.initialize(d, [:Grad, :Jac, :Hess])
+    d.disable_2ndorder ? MathProgBase.initialize(d, [:Grad, :Jac]) : MathProgBase.initialize(d, [:Grad, :Jac, :Hess])
```

Comment:
Use `:Hess in MathProgBase.features_available(d)` instead of `d.disable_2ndorder`. The former is the documented API (https://mathprogbasejl.readthedocs.io/en/latest/nlp.html#features_available) and the latter is an internal implementation detail that works only when `d` is of a certain type. This should resolve the test failures.